File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.swift
// Int N (immutable): An integer that represents length of Ai
if let input: String = readLine(), let N: Int = Int(input) {
    // [Int] Ai (immutable): Array of integers that stores integers to find LIS
    let Ai: [Int] = (readLine() ?? "1").split(separator: " ").compactMap { Int($0) }

    // [Int] dp (mutable): Array of integers to store LIS at each index
    var dp = [Int](repeating: 1, count: N)

    // Int gMax (mutable): An integer to update maximum length of LIS
    var gMax: Int = 1
    
    for i in 1..<N {
        for j in 0..<i {
            if Ai[j] < Ai[i] {
                // Update dp[i] to a larger value between 1 and dp[j]+1
                dp[i] = max(dp[i], dp[j]+1)
            }
        }
        // Update gMax
        gMax = max(gMax, dp[i])
    }
    
    print(gMax)
}

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
N = int(input())
Ai = list(map(int, input().split()))

# DP 배열 (최대 길이를 저장)
dp = [1] * N  

gMax = 1  # LIS의 최대 길이

for i in range(1, N):
    for j in range(i):  # i번째 원소보다 앞에 있는 원소 탐색
        if Ai[j] < Ai[i]:  # 증가하는 관계를 만족해야 함
            dp[i] = max(dp[i], dp[j] + 1)
    
    gMax = max(gMax, dp[i])  # 최댓값 업데이트

print(gMax)